docs/ci/ci-code-coverage-reporting.md

```diff
@@ -0,0 +1,138 @@
+# Code Coverage Reporting for InstructLab Repos
+
+## Problem Statement
+
+We currently capture code coverage reports for many of our repos with `pytest`, but that information is not directly relayed to pull request authors. To view code coverage reports, the pull request authors must sift through the CI logs to identify code coverage. Even then, the code coverage reporting only tells the contributor the overall code coverage -- not the coverage on their branch.
+
+## Target Goals
+
+By implementing code coverage reporting:
+
+* Contributors should be able to _easily_ identify which new lines of code in their branch need more coverage.
+  * This means contributors should not need to dig through CI logging to determine code coverage.
+  * Ideally, a bot should leave a comment to summarize the code coverage for them OR the contributor should be able to find this information without having to search through text (e.g., logging).
+* Contributors should understand how well their branch code is covered relative to the current `main` branch code.
+
+## High-Level Functional Requirements
+
+* Code coverage should be reported in PR builds as a GitHub _bot_ comment
+* New code:
+  * Should show % coverage on new lines of code.
+  * Should show which new lines of code are uncovered (if any).
+* Existing code:
+  * Should show existing % coverage on all lines of code.
+* New code + existing code:
+  * Should show projected % coverage if the PR is to be merged (e.g., “overall code coverage 53% -> 62% after merge”).
+* The GitHub coverage reporting bot should be able to interpret at least one of these pytest coverage reporting formats:
+  * HTML
+  * XML
+  * JSON
+  * lcov
+  * annotate
+* The README should display a code coverage badge
+
+## Code Coverage Reporting Bot: Options
+
+### [Codecov](https://github.com/marketplace/codecov)
+
+* [Example Codecov job](https://github.com/ansible/vscode-ansible/runs/33887209843)
+* Pros:
+  * Free to use in open source projects.
+  * Codecov reports all the data on GitHub in a comment. [Example](https://docs.codecov.com/docs/pull-request-comments).
+* Cons / Callouts:
+  * Not applicable at this time, but if we want to use it in >1 private repo, we must purchase a customer plan.
```

Comment:
We don't have any private repos (or plan to), so this shouldn't affect us.

Reply:
+1

Reply:
👍